google/gax/__init__.py

```diff
@@ -34,6 +34,8 @@
 import logging
 import multiprocessing as mp
 
+import dill as pickle
```

Comment:
This confused me also. I saw your manual `pickle.dumps` and `pickle.loads` calls and could not figure out why they were not `dill` calls. Turns out they are.

I think `import dill as pickle` is a totally valid way to address refactoring a large codebase, but in this case, I would maintain its nomenclature.